docs/installation.rst

```diff
@@ -67,11 +67,24 @@ Installation with NVIDIA GPU support (CUDA)
 
 .. note::
 
-   To use JAX on the GPU, make sure that your ``jaxlib`` is `built with CUDA support <https://github.com/google/jax#installation>`_.
+   There are 3 ways to install jax with CUDA support:
+   - using a pypi-distributed CUDA installation (suggested by jax developers) ``pip install -U 'jax[cuda12_pip]' -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.html`` 
+   - using the locally-installed CUDA version, which must be compatible with jax. ``pip install -U 'jax[cuda12_local]' -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.html`` 
+   The procedure to install ``mpi4jax`` for the two situations is different.
```

Comment:
We could recommend `--no-build-isolation` always for CUDA builds? Can't come up with a case where this would hurt to have.